rollup.config.js

```diff
@@ -4,44 +4,64 @@ import commonjs from '@rollup/plugin-commonjs';
 import replace from '@rollup/plugin-replace';
 import {terser} from 'rollup-plugin-terser';
 
-const config = mode => ({
-  input: 'src/Recoil_index.js',
-  output: {
-    file: `dist/recoil.${mode}.js`,
-    format: 'cjs',
-    exports: 'named',
-  },
-  external: ['react', 'react-dom'],
-  plugins: [
-    babel({
-      presets: ['@babel/preset-react', '@babel/preset-flow'],
+const modes = ['development', 'production'];
+const moduleFormats = ['cjs', 'esm'];
+
+const config = modes.flatMap(mode =>
+  moduleFormats.map(moduleFormat => {
+    let outputConfig;
+    switch (moduleFormat) {
+      case 'esm':
+        outputConfig = {
+          file: 'esm/recoil.${mode}.js',
```

Comment:
How do people use this esm output? `import Recoil from 'recoil/esm'`?

Reply:
Since I added "module" property to package.json, many bundle tools (e.g webpack) using module file inside esm directory automatically, so no update import statement is needed.